src/app/server/CommissioningWindowManager.cpp

```diff
@@ -454,6 +462,16 @@ CHIP_ERROR CommissioningWindowManager::StartAdvertisement()
         UpdateWindowStatus(CommissioningWindowStatusEnum::kBasicWindowOpen);
     }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD && CHIP_DEVICE_CONFIG_THREAD_FTD
+    // Block device role changing into Router if commissioning window opened and device not yet Router.
+    if (ConnectivityManagerImpl().GetThreadDeviceType() == ConnectivityManager::kThreadDeviceType_Router &&
+        otThreadGetDeviceRole(DeviceLayer::ThreadStackMgrImpl().OTInstance()) != OT_DEVICE_ROLE_ROUTER)
```

Comment:
`otThreadGetDeviceRole(DeviceLayer::ThreadStackMgrImpl().OTInstance()) != OT_DEVICE_ROLE_ROUTER` should be checked inside `SetRouterPromotion` (after taking the lock).

Reply:
Fixed